processing/src/main/java/io/druid/segment/IndexMerger.java

```diff
@@ -445,7 +445,9 @@ public int compare(Pair<Integer, PeekingIterator<String>> lhs, Pair<Integer, Pee
         }
         Indexed<String> indexed = dimValueLookups[i];
         if (useDirect) {
-          conversions[i] = ByteBuffer.allocateDirect(indexed.size() * Ints.BYTES).asIntBuffer();
+          int allocationSize = indexed.size() * Ints.BYTES;
+          log.info("Allocating dictionary merging direct buffer with size[%d]", allocationSize);
```

Comment:
There should also be a corresponding log message when the buffer is deallocated using `ByteBufferUtils.free(buf)`. And if it's never explicitly deallocated, then that is a problem! If that's the case, please either fix it or raise an issue for it.